apps/confidential/confidential.go

```diff
@@ -156,11 +156,20 @@ func NewCredFromSecret(secret string) (Credential, error) {
 }
 
 // NewCredFromAssertion creates a Credential from a signed assertion.
+//
+// Deprecated: a Credential created by this function can't refresh the
+// assertion when it expires. Use NewCredFromAssertionCallback instead.
 func NewCredFromAssertion(assertion string) (Credential, error) {
 	if assertion == "" {
 		return Credential{}, errors.New("assertion can't be empty string")
 	}
-	return Credential{assertion: assertion}, nil
+	return NewCredFromAssertionCallback(func(context.Context) (string, error) { return assertion, nil }), nil
```

Comment:
What data is available in the `Context`? TokenEndpoint (used as `aud` claim in the jwt) and `ClientId` (used as `iss` claim in the JWT) should be there to help app developer create the JWT.

Reply:
The `Context` is used to control the lifetime of IO operation.  If we need to pass additional data to the callback it should be in explicit parameters.